sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -108,6 +122,6 @@ private[sql] case class JDBCRelation(
   override def insert(data: DataFrame, overwrite: Boolean): Unit = {
     data.write
       .mode(if (overwrite) SaveMode.Overwrite else SaveMode.Append)
-      .jdbc(url, table, properties)
+      .jdbc(url, table, JDBCRelation.getEffectiveProperties(properties))
```

Comment:
This call is effectively just cloning the properties, but we're already doing the clone inside of `.jdbc()` itself, so we don't need this.